### PR TITLE
RFC:  introduce RFC process

### LINKED ISSUE #34: clarifications on rfc process
Migrated from Private Gitlab. Original Author Daniel Chen (@chenbh)

- What's the criteria to designate something as an RFC vs regular discussion? What are some examples of a substantial change that should be discussed in an RFC instead of slack/jira/gitlab issues/zoom call?
- What's the process after submitting an RFC? Who's reviewing it and how many reviewers can be expected? 
- What does it mean for an RFC to be merged? Does it signal that the team is comfortable with the idea, but doesn't have any responsibility to actually implement it; or does it turn into an item in the backlog that the team will eventually pick up?
- How does editing merged RFCs work (follow ups, modifications, deprecations)?

I've seen a couple of places where this whole process is documented as `RFC-0000` or `RFC-0001`: [python](https://www.python.org/dev/peps/pep-0001/), [kubernetes](https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/0000-kep-process/README.md), [rust](https://github.com/rust-lang/rfcs/blob/master/text/0002-rfc-process.md)

### LINKED ISSUE #540: Define the process for accepting RFCs

### PR DESCRIPTION
**Pull request by martyspiewak**
[Readable](https://github.com/vmware-tanzu/cartographer/blob/rfc-0019-rfc-process/rfc/rfc-0019-rfc-process.md)

Closes #540

### CODE REVIEW

**Comment by martyspiewak** [comment]
PR #554 implements the rest of the changes needed in the RFC dir to achieve the process defined in this RFC.

(Just cleanup of old templates and updating README)

**Comment by squeedee** [comment]
I really like this. I have no objections to ratifying it.

**Comment by emmjohnson** [comment]
Looks good to me. Thanks for writing this up Marty.

**Comment by idoru** [comment]
👍 LGTM. Could changing CRDs be one test for a "substantial change"?

**Comment by martyspiewak** [comment]
> 👍 LGTM. Could changing CRDs be one test for a "substantial change"?

@idoru I think that's what I was trying to capture with the first 2 bullet points:
```
* changes to a resource's spec
* changes to a resource's status
```
is there something else that we should capture for changing a CRD?

**Comment by idoru** [comment]
Yes, those bullets capture it well; is it still an unresolved question then?

**Comment by martyspiewak** [comment]
I guess not- just wanted to leave it open in case anyone had other thoughts for what should be in there. But if not, happy to remove that question.

**Comment by emmjohnson** [comment]
APPROVE

**Review by emmjohnson — APPROVED** [review]

File: rfc/rfc-0019-rfc-process.md
Comment (waciumawanjohi) [inline rfc/rfc-0019-rfc-process.md]:
```suggestion
- It will be discussed in a future office hours meeting. Office hours happen on a weekly cadence barring exceptions.
```

David has hoped to keep RFCs to Office Hours

File: rfc/rfc-0019-rfc-process.md
Comment (waciumawanjohi) [inline rfc/rfc-0019-rfc-process.md:64]: Say 5 maintainers abstain, 2 vote Affirmative and 1 votes Negative.

Do the Affirmatives carry the day?

Comment (waciumawanjohi) [inline rfc/rfc-0019-rfc-process.md:64]: Say 5 maintainers abstain, 2 vote Affirmative and 1 votes Negative.

Do the Affirmatives carry the day?

From a Robert's Rules of Order explainer:
> Abstentions are instances in which members who are present refuse to vote. In the usual
situation where either a majority vote or a two-thirds vote is required, abstentions are not counted
and have no effect on the result. However, if the vote required is a majority or two-thirds of the
members present, an abstention has the same effect as a “no” vote.

File: rfc/rfc-0019-rfc-process.md
Comment (waciumawanjohi) [inline rfc/rfc-0019-rfc-process.md:65]: An RFC is in FCP. A maintainer raises a new concern. Does the RFC go into development automatically? Does the maintainer open a second motion for the RFC to go into development?

This sounds like a process that we'll need to specify now or figure out ad-hoc.

File: rfc/rfc-0019-rfc-process.md
Comment (waciumawanjohi) [inline rfc/rfc-0019-rfc-process.md]: Say 5 maintainers abstain, 2 vote Affirmative and 1 votes Negative.

Do the Affirmatives carry the day?

From a Robert's Rules of Order explainer:

> Abstentions are instances in which members who are present refuse to vote. In the usual
situation where either a majority vote or a two-thirds vote is required, abstentions are not counted
and have no effect on the result. However, if the vote required is a majority or two-thirds of the
members present, an abstention has the same effect as a “no” vote.

File: rfc/rfc-0019-rfc-process.md
Comment (waciumawanjohi) [inline rfc/rfc-0019-rfc-process.md:65]: An RFC is in FCP. A maintainer raises a new concern. Does the RFC go into development automatically? Does the maintainer open a second motion for the RFC to go into development?

This sounds like a process that we'll need to specify now or figure out ad-hoc.

File: rfc/rfc-0019-rfc-process.md
Comment (waciumawanjohi) [inline rfc/rfc-0019-rfc-process.md:88]: An additional drawback:
In not archiving rejected/unapproved RFCs, Carto may lose context of past discussions and decisions. Sometimes a decision not to do X can be as significant as the decision to do Y.

File: rfc/rfc-0019-rfc-process.md
Comment (waciumawanjohi) [inline rfc/rfc-0019-rfc-process.md]: I think you've answered these questions in the RFC, I'm happy with those answers.

File: rfc/rfc-0019-rfc-process.md
Comment (martyspiewak) [reply rfc/rfc-0019-rfc-process.md]: Above it says:
> Anyone who chooses not to actively partcipate in any RFC is presumed to trust their colleagues on the matter.

I had assumed based on that that yes, the 2 affirmatives would carry the day. Do you think it should be different? Or that this should be called out explicitly?

File: rfc/rfc-0019-rfc-process.md
Comment (martyspiewak) [reply rfc/rfc-0019-rfc-process.md:65]: I think we can figure that out ad-hoc. If a maintainer raises such a significant concern that others are swayed, i think this will likely cause it to go back into development. If others disagree and quickly convince the maintainer it isn't a concern, it will stay in FCP.

File: rfc/rfc-0019-rfc-process.md
Comment (martyspiewak) [reply rfc/rfc-0019-rfc-process.md:88]: That's fair. I think those could be found by looking at past PRs with the `rfc` label so we wouldn't lose that context but I hear it's a little less visible.

**Review by squeedee — APPROVED** [review]

File: rfc/rfc-0019-rfc-process.md
Comment (waciumawanjohi) [reply rfc/rfc-0019-rfc-process.md]: I assumed the same as you, and wanted to get that captured in writing so that any future disagreement would have a paper trail to provide resolution.

File: rfc/rfc-0019-rfc-process.md
Comment (waciumawanjohi) [inline rfc/rfc-0019-rfc-process.md:112]:
```suggestion
```

Comment (waciumawanjohi) [inline rfc/rfc-0019-rfc-process.md]:
```suggestion
```

File: rfc/rfc-0019-rfc-process.md
Comment (waciumawanjohi) [inline rfc/rfc-0019-rfc-process.md:88]:
```suggestion
In not archiving rejected/unapproved RFCs, Cartographer may lose context of past discussions and decisions.

```

File: rfc/rfc-0019-rfc-process.md
Comment (waciumawanjohi) [reply rfc/rfc-0019-rfc-process.md]:
```suggestion
- For voting, the binding votes are comprised of the maintainer team. Acceptance requires a super majority of binding votes in favor. The voting options are the following: Affirmative, Negative, and Abstain. Non-binding votes are of course welcome. Super majority means 2/3 or greater of binding votes. (Abstentions are not counted in the calculation)
 ```

**Review by jwntrs — APPROVED** [review]

**Comment by davidmirror-ops** [comment]
Also I believe it closes #34

**Review by cirocosta — APPROVED** [review]

**Comment by cirocosta** [comment]
some thoughts from community call: making use of [code owners](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) we could more easily tag a group of folks that _must_ provide their feedback

**Review by waciumawanjohi — APPROVED** [review]

File: rfc/rfc-0019-rfc-process.md
Comment (sclevine) [inline rfc/rfc-0019-rfc-process.md]: I want to suggest one more step here: at least one of @sclevine, @rawlingsj, or @scothis must +1 for the RFC to be approved. (Eventually, this group could be a named committee with more inclusive membership.)

**Review by sclevine — CHANGES_REQUESTED** [review]

File: rfc/rfc-0019-rfc-process.md
Comment (cirocosta) [reply rfc/rfc-0019-rfc-process.md]: @sclevine wdyt of making use of GitHub teams for that?

Screen Shot 2022-02-07 at 2 09 09 PM

e.g., have a team that's composed of @sclevine , @rawlingsj , @scothis as part of it so we could block merge based on that team's approval

Comment (martyspiewak) [reply rfc/rfc-0019-rfc-process.md]: Updated- I referred to this as the steering committee, but happy to change if there's a better name.

File: rfc/rfc-0019-rfc-process.md
Comment (scothis) [inline rfc/rfc-0019-rfc-process.md]: RFCs should be timeless. It's better to create a team with the individuals listed and reference that team from here.

Comment (martyspiewak) [reply rfc/rfc-0019-rfc-process.md]: Fixed, thanks.

File: rfc/rfc-0019-rfc-process.md
Comment (sclevine) [inline rfc/rfc-0019-rfc-process.md]:
```suggestion
- For voting, the binding votes are comprised of the maintainer team and Technical Oversight Committee. Acceptance requires a super majority of maintainer votes and at least one Technical Oversight Committee vote in favor. The voting options are the following: Affirmative, Negative, and Abstain. Non-binding votes are of course welcome. Super majority means 2/3 or greater of binding votes. (Abstentions are not counted in the calculation)
```

**Review by sclevine — CHANGES_REQUESTED** [review]

**Review by sclevine — APPROVED** [review]

**Review by scothis — APPROVED** [review]

**Pull request merged**
